internal/access/grant.go

```diff
@@ -30,7 +30,15 @@ func ListGrants(c *gin.Context, subject uid.PolymorphicID, resource string, priv
 	roles := []string{models.InfraAdminRole, models.InfraViewRole, models.InfraConnectorRole}
 	db, err := RequireInfraRole(c, roles...)
 	if err == nil {
-		selectors = append(selectors, data.ByOptionalSubject(subject))
+		if inherited && subject.IsIdentity() {
```

Comment:
I think what Bruce was trying to get at here is that this only looks at `inherited` if the request is being authorized by a grant.

If the request is being authorized because the request is for the user's own grants, then the `inherited` flag is being ignored.

I think this logic needs to happen above, so that the selector is added before going into this `err == nil` block.

Reply:
got it, thanks. added a test case for this.